blockprod/src/detail/tests.rs

```diff
@@ -400,291 +1411,374 @@ mod produce_block {
+                        .produce_block_with_custom_id(
+                            GenerateBlockInputData::None,
+                            TransactionsSource::Provided(vec![]),
+                            Some(id.clone()),
+                        )
+                        .await;
+
+                    match result {
+                        Err(BlockProductionError::JobManagerError(
+                            JobManagerError::JobAlreadyExists,
+                        )) => break,
+                        Err(_) => panic!("Duplicate job key should fail"),
+                        Ok(_) => continue,
+                    }
+                }
+            }
+        });
 
-    let jobs_count = block_production.job_manager_handle.get_job_count().await.unwrap();
-    assert_eq!(jobs_count, 0, "Jobs count is incorrect");
+        manager.main().await;
+        join_handle.await.unwrap();
+    }
 }
 
-#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn stop_all_jobs_error() {
-    let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test(None);
+mod stop_all_jobs {
+    use super::*;
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn error() {
+        let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test(None);
 
-    let mut block_production = BlockProduction::new(
-        chain_config,
-        chainstate.clone(),
-        mempool,
-        Default::default(),
-        prepare_thread_pool(1),
-    )
-    .expect("Error initializing blockprod");
+        let mut block_production = BlockProduction::new(
+            chain_config,
+            chainstate.clone(),
+            mempool,
+            Default::default(),
+            prepare_thread_pool(1),
+        )
+        .expect("Error initializing blockprod");
 
-    let mut mock_job_manager = Box::<MockJobManager>::default();
+        let mut mock_job_manager = Box::<MockJobManager>::default();
 
-    mock_job_manager
-        .expect_stop_all_jobs()
-        .times(1)
-        .returning(|| Err(JobManagerError::FailedToStopJobs));
+        mock_job_manager
+            .expect_stop_all_jobs()
+            .times(1)
+            .returning(|| Err(JobManagerError::FailedToStopJobs));
 
-    block_production.set_job_manager(mock_job_manager);
+        block_production.set_job_manager(mock_job_manager);
 
-    let result = block_production.stop_all_jobs().await;
+        let result = block_production.stop_all_jobs().await;
 
-    match result {
-        Err(BlockProductionError::JobManagerError(_)) => {}
-        _ => panic!("Unexpected return value"),
+        match result {
+            Err(BlockProductionError::JobManagerError(_)) => {}
+            _ => panic!("Unexpected return value"),
+        }
+    }
+
+    #[rstest]
+    #[trace]
+    #[case(Seed::from_entropy())]
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn ok(#[case] seed: Seed) {
```

Comment:
A more descriptive test name would be appreciated. Something that suggests what is expected to be ok.

Reply:
Yep cool. I thought it would be ok since it's in a separate `mod` per function, but no problem.

Reply:
If it's in a separate mod, that's fine, I'm ok with it either way. It's not obvious when just scrolling through the code though.